apps/autorelayer-interop/src/relayer.ts

```diff
@@ -12,24 +12,38 @@ import { z } from 'zod'
 
 import { jsonFetchParams } from '@/utils/jsonFetchParams.js'
 
-const PendingMessagesSchema = z.array(
-  z.object({
-    // Identifier
-    messageHash: z.string().refine(isHex, 'invalid message hash'),
-    // Message Direction
-    source: z.number(),
-    destination: z.number(),
-    // ExecutingMessage
-    logIndex: z.number(),
-    logPayload: z.string().refine(isHex, 'invalid log payload'),
-    timestamp: z.number(),
-    blockNumber: z.number(),
-    transactionHash: z.string().refine(isHash, 'invalid transaction hash'),
-  }),
-)
+const PendingMessageSchema = z.object({
+  // Identifier
+  messageHash: z.string().refine(isHex, 'invalid message hash'),
+  // Message Direction
+  source: z.number(),
+  destination: z.number(),
+  // ExecutingMessage
+  logIndex: z.number(),
+  logPayload: z.string().refine(isHex, 'invalid log payload'),
+  timestamp: z.number(),
+  blockNumber: z.number(),
+  transactionHash: z.string().refine(isHash, 'invalid transaction hash'),
+})
+
+const PendingMessagesSchema = z.array(PendingMessageSchema)
+
+export type PendingMessage = z.infer<typeof PendingMessageSchema>
 
 type PendingMessages = z.infer<typeof PendingMessagesSchema>
 
+/**
+ * Interface for pending message validation strategies
+ */
+export interface PendingMessageValidationStrategy {
```

Comment:
instead of creating this strategy interface that's passed in the constructor, can we add a protected method that can be implemented?

```typescript
class Relayer {

// default unimplemented
protected async validate(message: PendingMessage): Promise<void> = {}
```

then when extending the class, we can just define the method